enginetest/insert_queries.go

```diff
@@ -328,6 +328,78 @@ var InsertQueries = []WriteQueryTest{
 			{10, "hello"},
 		},
 	},
+	{
+		"INSERT INTO auto_increment_tbl (c0) values (44)",
+		[]sql.Row{{sql.NewOkResult(1)}},
+		"SELECT * FROM auto_increment_tbl ORDER BY pk",
+		[]sql.Row{
+			{1, 11},
+			{2, 22},
+			{3, 33},
+			{4, 44},
+		},
+	},
+	{
+		"INSERT INTO auto_increment_tbl (c0) values (44),(55)",
+		[]sql.Row{{sql.NewOkResult(2)}},
+		"SELECT * FROM auto_increment_tbl ORDER BY pk",
+		[]sql.Row{
+			{1, 11},
+			{2, 22},
+			{3, 33},
+			{4, 44},
+			{5, 55},
+		},
+	},
+	{
+		"INSERT INTO auto_increment_tbl values (NULL, 44)",
+		[]sql.Row{{sql.NewOkResult(1)}},
+		"SELECT * FROM auto_increment_tbl ORDER BY pk",
+		[]sql.Row{
+			{1, 11},
+			{2, 22},
+			{3, 33},
+			{4, 44},
+		},
+	},
+	{
+		"INSERT INTO auto_increment_tbl values (0, 44)",
+		[]sql.Row{{sql.NewOkResult(1)}},
+		"SELECT * FROM auto_increment_tbl ORDER BY pk",
+		[]sql.Row{
+			{1, 11},
+			{2, 22},
+			{3, 33},
+			{4, 44},
+		},
+	},
+	{
+		"INSERT INTO auto_increment_tbl values (5, 44)",
```

Comment:
Should have another test for inserting a value lower than the current max value (need sparse values in the table for this)